cmd/geth/main.go

```diff
@@ -347,6 +351,15 @@ func geth(ctx *cli.Context) error {
 	stack, backend := makeFullNode(ctx)
 	defer stack.Close()
 
+	if ctx.GlobalBool(utils.RunHeimdallFlag.Name) {
+		shutdownCtx, stop := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+		defer stop()
+
+		go func() {
+			service.NewHeimdallService(shutdownCtx, getHeimdallArgs(ctx))
+		}()
+	}
+
 	startNode(ctx, stack, backend, false)
 	stack.Wait()
```

Comment:
@JekaMas Here, If I am not wrong this will keep the process running until it's supposed to terminate.